qcodes/instrument_drivers/Lakeshore/Model_325.py

```diff
@@ -278,41 +277,14 @@ def __init__(self, parent: 'Model_325', name: str, inp: str) -> None:
             vals=Numbers(min_value=1, max_value=35)
         )
 
-    def decode_sensor_status(self, sum_of_codes: int) -> str:
-        """
-        The sensor status is one of the status codes, or a sum thereof. Multiple
-        status are possible as they are not necessarily mutually exclusive.
-
-        args:
-            sum_of_codes (int)
-        """
-        components = list(self.sensor_status_codes.keys())
-        codes = self._get_sum_terms(components, int(sum_of_codes))
-        return ", ".join([self.sensor_status_codes[k] for k in codes])
-
     @staticmethod
-    def _get_sum_terms(components: list, number: int):
-        """
-        Example:
-        >>> components = [0, 1, 16, 32, 64, 128]
-        >>> _get_sum_terms(components, 96)
-        >>> ...[64, 32]  # This is correct because 96=64+32
-        """
-        if number in components:
-            terms = [number]
-        else:
-            terms = []
-            comp = np.sort(components)[::-1]
-            comp = comp[comp <= number]
-
-            while len(comp):
-                c = comp[0]
-                number -= c
-                terms.append(c)
-
-                comp = comp[comp <= number]
-
-        return terms
+    def decode_sensor_status(sum_of_codes: int) -> str:
+        total_status = Status(int(sum_of_codes))
```

Comment:
what is the conversion to `int` needed for here if the input type is already `int`?
```suggestion
        total_status = Status(sum_of_codes)
```

Reply:
The input from the machine was actually `str`, not `int`. I changed the `get_parser` of the parameter `status` to a lambda function `get_parser=lambda status: self.decode_sensor_status(int(status))`, so that the parser can convert the input to int and call the decoder.